File: baseproject-android/app/src/main/java/com/chang/template/activity/BaseFragment.java
package com.chang.template.activity;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.afollestad.materialdialogs.MaterialDialog;
import com.chang.template.R;
import com.chang.template.utils.Utility;

import butterknife.ButterKnife;

/**
 * Created by Howard Chang on 2016/11/29
 */
public abstract class BaseFragment extends Fragment implements BaseView {

    protected ProgressDialog progressDialog;

    @LayoutRes
    protected abstract int getContentView();


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(getContentView(), container, false);
        ButterKnife.bind(this, v);

        progressDialog = Utility.newProgressDialog(getActivity());

        return v;
    }


    @Override
    public void onDestroy() {

        if (progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
        super.onDestroy();
    }

    @Override
    public void onInternetError() {
        progressDialog.dismiss();
        new MaterialDialog.Builder(getActivity())
                .title(R.string.app_name)
                .content(R.string.g_a_network_error)
                .positiveText(R.string.g_confirm)
                .show();
    }

    @Override
    public void onRequestEnd() {
        progressDialog.dismiss();
    }

    @Override
    public void onRequestError(String msg) {
        progressDialog.dismiss();
        new MaterialDialog.Builder(getActivity())
                .title(R.string.app_name)
                .content(msg)
                .positiveText(R.string.g_confirm)
                .show();
    }

    @Override
    public void onRequestStart() {
        progressDialog.show();
    }

    @Override
    public void onRequestMessage(String msg) {
        new MaterialDialog.Builder(getActivity())
                .content(msg)
                .positiveText(R.string.g_ok)
                .show();
    }

}

File: baseproject-android/app/src/main/java/com/chang/template/activity/BaseFragment.kt
package com.chang.template.activity

import android.app.ProgressDialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment
import com.afollestad.materialdialogs.MaterialDialog
import com.chang.template.R
import com.chang.template.activity.home.MainFragment
import com.chang.template.utils.Utility

/**
 * Created by Howard Chang on 2016/11/29
 */
abstract class BaseFragment : Fragment(), BaseView {

    protected var progressDialog: ProgressDialog? = null

    @get:LayoutRes
    protected abstract val contentView: Int

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(contentView, container, false)

        activity?.apply {
            progressDialog = Utility.newProgressDialog(this.applicationContext)
        }

        return v
    }

    override fun onDestroy() {
        progressDialog?.apply {
            if (this.isShowing) {
                this.dismiss()
            }
        }

        super.onDestroy()
    }

    override fun onInternetError() {
        progressDialog?.dismiss()
        activity?.apply {
            MaterialDialog(this).show {
                title(R.string.app_name)
                message(R.string.g_a_network_error)
                positiveButton(R.string.g_confirm)
            }
        }
    }

    override fun onRequestEnd() {
        progressDialog?.dismiss()
    }

    override fun onRequestError(msg: String) {
        progressDialog?.dismiss()
        activity?.apply {
            MaterialDialog(this).show {
                title(R.string.app_name)
                message(text = msg)
                positiveButton(R.string.g_confirm)
            }
        }
    }

    override fun onRequestStart() {
        progressDialog?.show()
    }

    override fun onRequestMessage(msg: String) {
        activity?.apply {
            MaterialDialog(this).show {
                title(R.string.app_name)
                message(text = msg)
                positiveButton(R.string.g_confirm)
            }
        }
    }

    companion object {
        const val DATA = "fragment_data"
    }
}
